### PR TITLE
request opts

### PR DESCRIPTION
**Pull request by TopiSenpai**
this aims to make it possible to add modifiers to requests like `context.Context`, `X-Audit-Log-Reason` and possible other stuff in the future

Additionally usecases:
* Checks
* Request delay

### CODE REVIEW

**Comment by TopiSenpai** [comment]
this should be ready to review @Skye-31 
just ignore missing docs ty :D

File: _examples/webhook/example.go [hunk 1]
```diff
@@ -2,29 +2,59 @@ package main
...
+	<-s
+}
 
-	cancel()
+func send(ctx context.Context, client webhook.Client, i int) {
+	_, err := client.CreateMessage(webhook.NewMessageCreateBuilder().SetContentf("test %d", i).Build(),
+		rest.WithCtx(ctx),
+		rest.WithReason("this adds a reason header"),
+		rest.WithDelay(2*time.Second),
+		rest.WithCheck(func() bool {
```
Comment (Skye-31) [inline _examples/webhook/example.go:53]: What's the purpose of this?

File: core/entity_builder_impl.go [hunk 2]
```diff
@@ -278,34 +278,6 @@ func (b *EntityBuilderImpl) CreateGuild(guild discord.Guild, updateCache CacheSt
...
-	for _, emote := range guild.Emojis {
-		b.CreateEmoji(guild.ID, emote, updateCache)
-	}
-
-	// TODO: presence
-	/*for i := range fullGuild.Presences {
-		presence := fullGuild.Presences[i]
-		presence.Disgo = disgo
-		b.Disgo().Cache().CachePresence(presence)
-	}*/
```
Comment (Skye-31) [inline core/entity_builder_impl.go:308]: Was all of this logic meant to be removed?

File: discord/gateway_intents.go [hunk 3]
```diff
@@ -42,6 +42,9 @@ const (
 	GatewayIntentsAll = GatewayIntentsNonPrivileged |
 		GatewayIntentsPrivileged
 
+	GatewayIntentsDefault = GatewayIntentGuilds |
```
Comment (Skye-31) [inline discord/gateway_intents.go:45]: I don't think it's good practice to provide defaults for Intents, it's something that should be explicitly set by the user.

**Comment by Skye-31** [comment]
Left some questions

File: core/entity_builder_impl.go (hunk 2 above)
Comment (TopiSenpai) [reply core/entity_builder_impl.go:308]: yep I moved it into the gateway event handler

File: discord/gateway_intents.go (hunk 3 above)
Comment (TopiSenpai) [reply discord/gateway_intents.go:45]: then default would be `0`
idk if thats better

File: _examples/webhook/example.go (hunk 1 above)
Comment (TopiSenpai) [reply _examples/webhook/example.go:53]: the checks are executed after the `delay` & `ratelimit` was awaited if necessary

those can be used to easily do last minute checks

File: discord/gateway_intents.go (hunk 3 above)
Comment (Skye-31) [reply discord/gateway_intents.go:45]: I disagree, imo it should throw an error if not provided

File: core/entity_builder_impl.go (hunk 2 above)
Comment (Skye-31) [reply core/entity_builder_impl.go:308]: Ah, ok

File: discord/gateway_intents.go (hunk 3 above)
Comment (TopiSenpai) [reply discord/gateway_intents.go:45]: that would require it to be a pointer which is ugly >.>

Comment (Skye-31) [reply discord/gateway_intents.go:45]: then defaulting to 0 is better

Comment (TopiSenpai) [reply discord/gateway_intents.go:45]: oki

**Pull request merged**